src/support/generation.c

```diff
@@ -53,7 +53,7 @@ __wt_gen_init(WT_SESSION_IMPL *session)
      * All generations start at 1, a session with a generation of 0 isn't using the resource.
      */
     for (i = 0; i < WT_GENERATIONS; ++i)
-        S2C(session)->generations[i] = 1;
```

Comment:
This happens once and not in a concurrent context, is atomic required?

Reply:
If TSan flags a variable as data racy then I'm changing every instance of the variable to make them atomic. 
This keeps all accesses consistent and will make life easier for us if/when we can migrate to using `_Atomic`